### ISSUE TITLE
plugin-cloud-storage: TypeError: options.adapter is not a function with Vite Bundler

### ISSUE THREAD

**Issue by paper-krane** [payloadcms/payload#4843]
For some reason the cloud storage plugin (version 1.1.1) is spitting out an error to the client side terminal. The Vite bundler (version 0.1.2) isn't catching this error when compiling. But on the client side, it is not rendering Payload's views and is spitting the error in the console:

```
Uncaught (in promise) TypeError: options.adapter is not a function
    at plugin.ts:43:35
    at Array.map (<anonymous>)
    at plugin.ts:39:47
    at build.ts:16:14
    at async buildConfig (build.ts:14:32)
```

I didn't have this issue with Webpack however Webpack has been giving me issues with other aspects of the small app I am building. My current adapter is going to Digital Oceans S3 storage and looks like: 

```js
import { s3Adapter } from '@payloadcms/plugin-cloud-storage/s3';

export const s3Adaptation = s3Adapter({
    config: {
        endpoint: process.env.S3_SUPER_SECRET_ENDPOINT,
        region: process.env.S3_SUPER_SECRET_REGION,
        credentials: {
            accessKeyId: process.env.S3_SUPER_SECRET_ACCESS_KEY,
            secretAccessKey: process.env.S3_SUPER_SECRET_SECRET_KEY,
        },
    },
    acl: 'public-read',
    bucket: 'my-bucket-name',
});
```

And the config file looks like:

```js
import { buildConfig } from 'payload/config';

// Plugins
import { cloudStorage } from '@payloadcms/plugin-cloud-storage';
import { s3Adapter } from '@payloadcms/plugin-cloud-storage/s3';

// Adapters
import { s3Adaptation } from './utilities/s3Storage';

export default buildConfig({
    // ... Config options
    plugins: [
        cloudStorage({
	    collections: {
		'media': {
		    adapter: s3Adaptation,
		    prefix: 'Public',
		},
		'files': {
		    adapter: s3Adaptation,
		    prefix: 'Secondary',
		}
	    }
	})
    ],
})
```

Thank you so much for taking a look at this! Hopefully I didn't break anything on my end.

**Comment by Enot-Racoon** [comment]
Pls help. How to fix HMR after migrate on Vite bundler ? https://github.com/payloadcms/payload/issues/4671#issuecomment-1900838913

**Comment by AlessioGr** [comment]
This issue has been fixed in 3.0. Previously, plugin-cloud-storage was using webpack aliases in order to ensure server-only functionality does not make it to the client. This was likely not properly handled for vite.

In 3.0 we have decided to rely on next.js to handle the server-client boundary, and we have completely dropped support for our vite (and webpack) bundler. The best course of action here would be to upgrade to 3.0
